src/background.ts

```diff
@@ -0,0 +1,284 @@
+  "authDomain": "moz-fx-data-rally-w-prod-dfa4.firebaseapp.com",
+  "projectId": "moz-fx-data-rally-w-prod-dfa4",
+  "storageBucket": "moz-fx-data-rally-w-prod-dfa4.appspot.com",
+  "messagingSenderId": "982322764946",
+  "appId": "1:982322764946:web:f9b6aea488cebde47ada4b",
+  "functionsHost": "https://us-central1-moz-fx-data-rally-w-prod-dfa4.cloudfunctions.net"
+};
+
+// Overrides for dev mode - use local emulators with "exampleStudy1" as study ID.
+if (enableEmulatorMode) {
+  studyId = "exampleStudy1";
+  rallySite = "http://localhost:3000";
+  firebaseConfig = {
+    "apiKey": "abc123",
+    "authDomain": "demo-rally.firebaseapp.com",
+    "projectId": "demo-rally",
+    "storageBucket": "demo-rally.appspot.com",
+    "messagingSenderId": "abc123",
+    "appId": "1:123:web:abc123",
+    "functionsHost": "http://localhost:5001"
+  }
+}
+
+// This function will be called when the study state changes. By default,
+// a study starts "paused". If a user opts-in to a particular study, then the
+// state will change to "started".
+//
+// The study state may change at any time (for example, the server may choose to pause a particular study).
+// Studies should stop data collection and try to unload as much as possible when in "paused" state.
+async function stateChangeCallback(newState) {
+  switch (newState) {
+    case (RunStates.Running):
+      // The all-0 Rally ID indicates developer mode, in case data is accidentally sent.
+      let rallyId = enableDevMode ? "00000000-0000-0000-0000-000000000000" : rally.rallyId;
+
+      // The all-1 Rally ID means that there was an error with the Rally ID.
+      if (!rallyId) {
+        rallyId = "11111111-1111-1111-1111-111111111111";
+      }
+      console.info(`Study running with Rally ID: ${rallyId}`);
+
+      const storage = await browser.storage.local.get("enrolled");
+      if (storage.enrolled !== true) {
+        console.info("Recording enrollment.");
+        rallyManagementMetrics.id.set(rallyId);
+        examplePings.studyEnrollment.submit();
+
+        browser.storage.local.set({
+          enrolled: true,
+        });
+      }
+
+      // The Rally API has been initialized.
+      // Initialize the study and start it.
+
+      // Example: initialize the example module.
+      exampleModuleInitialize();
+      await browser.storage.local.set({ "state": RunStates.Running });
+
+
+      // Example: set a listener for WebScience page navigation events on
+      // http://localhost/* pages. Note that the manifest origin
+      // permissions currently only include http://localhost/*. You should
+      // update the manifest permissions as needed for your study.
+
+      this.pageDataListener = async (pageData) => {
+        console.log(`WebScience page navigation event fired with page data:`, pageData);
+        userJourney.pageId.set(pageData.pageId);
+        // Round attention duration up to 1ms, Glean will balk if it is lower than this.
+        if (pageData.attentionDuration < 1) {
```

Comment:
I remember seeing the discussion about this in Slack.

After doing some testing, I think the problem here isn't actually with "sub-1" duration, but actually with any non-integer duration. What I see happening is that usually the WebScience event outputs an integer `attentionDuration`, but occasionally it instead output a float. I haven't figured out how to reproduce it yet, but if you generate a bunch of varied page navigations you should see it. I think the occasional float is a bug with WebScience.

Anyways, we can either dig into the underlying WebScience bug, or we can just Math.floor our attentionDuration.

Also, everything above ditto applies to audioDuration; I see the same behavior if I spam click through YouTube for a bit.

Example of integer attentionDuration:
<img width="610" alt="Screen Shot 2022-06-15 at 12 47 27 AM" src="https://user-images.githubusercontent.com/7362472/173670837-cc1692c7-2adb-4415-9d17-c0c1b093c5fb.png">

Example of float attentionDuration:
<img width="605" alt="Screen Shot 2022-06-15 at 12 47 55 AM" src="https://user-images.githubusercontent.com/7362472/173670896-e72f245c-bd2a-44ea-bbc1-850b618222d4.png">

Example of integer audioDuration:
<img width="605" alt="Screen Shot 2022-06-15 at 12 48 19 AM" src="https://user-images.githubusercontent.com/7362472/173670963-990dbbaa-01db-415f-98c0-43146133e974.png">

Example of float audioDuration:
<img width="610" alt="Screen Shot 2022-06-15 at 12 48 35 AM" src="https://user-images.githubusercontent.com/7362472/173671018-6ecbf822-dac1-42b8-8363-70deada786c6.png">

(these are all from the same "journey")

Reply:
Thanks for digging into this! That makes sense. I am not sure who is really at "fault" here, WebScience or Glean or the study-template... on the JS side this is presumably a [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number), which is a `double` so it can represent both whole numbers and decimals.

Glean however wants this to be an integer, so I think doing `Math.floor()` might make the most sense - WebScience returning a `Number` to JS callers is fine, it's really Glean that has the requirement that this be an integer, so I think it makes sense for the study code to be responsible for this.

Reply:
We handle similar cases below, such as how scroll depth and datetimes are handled in JS vs. Glean.